contributed_definitions/nyaml/NXxps.yaml

```diff
@@ -0,0 +1,520 @@
+            * near ambient pressure X-ray photoelectron spectroscopy (NAPXPS)
+            * electron spectroscopy for chemical analysis (ESCA)
+        
+        .. _ISO 18115-1:2023: https://www.iso.org/standard/74811.html
+        .. _Clause 11: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:sec:11
+    geometries(NXcoordinate_system_set):
+      xps_coordinate_system(NXcoordinate_system):
+        exists: optional
+        doc: |
+          In traditional surface science, the coordinate system is defined such that the positive z-axis
+          points along the normal of the sample stage, and the x- and y-axes lie in the plane of the sample stage.
+          However, in NeXus, a coordinate system that is the same as `McStas`_  is used. `xps_coordinate_system`
+          gives the user the opportunity to work in the traditional base coordinate system.
+          
+          .. _McStas: http://mcstas.org/
+          
+          .. image:: xps/xps_cs.png
+              :width: 40%
+        origin:
+          enumeration: [sample stage]
+        handedness:
+          enumeration: [left_handed]
+        z_direction:
+          enumeration: [sample stage normal]
+        x:
+          enumeration: [[-1, 0, 0]]
+        y:
+          enumeration: [[0, 1, 0]]
+        z:
+          enumeration: [[0, 0, 1]]
+        depends_on(NX_CHAR):
+          doc: |
+            Link to transformations defining the XPS base coordinate system,
+            which is defined such that the positive z-axis points along the sample stage normal, and the
+            x- and y-axes lie in the plane of the sample stage.
+        coordinate_system_transformations(NXtransformations):
+          doc: |
+            Set of transformations, describing the orientation of the XPS coordinate system
+            with respect to the beam coordinate system (.).
+    (NXinstrument):
+      doc:
+      - |
+        Description of the XPS spectrometer and its individual parts.
+      - |
+        xref:
+          spec: ISO 18115-1:2023
+          term: 12.58
+          url: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.58
+      sourceTYPE(NXsource):
+        power(NX_FLOAT):
+          unit: NX_POWER
+          exists: recommended
+      beamTYPE(NXbeam):
+        depends_on:
+          exists: recommended
+          doc: |
+            Reference to the transformation describing the orientation of the beam
+            relative to a defined coordinate system.
+        transformations(NXtransformations):
+          exists: recommended
+          beam_azimuth_angle(NX_NUMBER):
+            unit: NX_ANGLE
+            doc: |
+              Azimuthal tilt of the beam from the y-direction towards the operator, defined
+              by the sample stage.
+            \@transformation_type:
+              enumeration: [rotation]
+            \@vector:
+              enumeration: [[0, 0, -1]]
```

Comment:
Normally, nexus rotates counter-clockwise around the axis for positive values (https://manual.nexusformat.org/design.html#the-nexus-coordinate-system). From the graphic I interpret that you also want a counter-clockwise rotation. So I think you want to use the positive z-axis here.

Reply:
Ok, I thought nexus was doing it clockwise. I changed it to positive z-axis values.

Reply:
For the sample rotation, this should then still be negative, correct?

Reply:
I find clockwise/counter-clockwise always a bit unspecific, because it depends on which way you look. The proper terms are right-handed/left-handed, I think. McSTAS is right-handed (which if you look along the defining axis corresponds to clockwise rotation, and to a counter-clockwise rotation if you look "into" the corresponding axis).

Reply:
> For the sample rotation, this should then still be negative, correct?

The sample rotation seems to be the only right-handed rotation in your definition, yes. But according to the discussion above, I would say all ref. axes should be negative apart from this one, or do I miss something?

Reply:
That would mean that the sense of "positive" rotation also changes with the handedness of the CO system, i.e. one would need other rotation matrices depending on the handedness of the system. Wikipedia says:
https://en.wikipedia.org/wiki/Rotation_(mathematics)

Rotation can have a [sign](https://en.wikipedia.org/wiki/Sign_(mathematics)) (as in the [sign of an angle](https://en.wikipedia.org/wiki/Angle#Sign)): a clockwise rotation is a negative magnitude so a counterclockwise turn has a positive magnitude.

I am not sure if this is compatible with what you propose.

Reply:
In three-dimensional geometry, "clockwise" and "anticlockwise" have no absolute meaning, so the direction of positive and negative angles must be defined in terms of an [orientation](https://en.wikipedia.org/wiki/Orientability), which is typically determined by a [normal vector](https://en.wikipedia.org/wiki/Normal_(geometry)) passing through the angle's vertex and perpendicular to the plane in which the rays of the angle lie.

In [navigation](https://en.wikipedia.org/wiki/Navigation), [bearings](https://en.wikipedia.org/wiki/Bearing_(navigation)) or [azimuth](https://en.wikipedia.org/wiki/Azimuth) are measured relative to north. By convention, viewed from above, bearing angles are positive clockwise, so a bearing of 45° corresponds to a north-east orientation. Negative bearings are not used in navigation, so a north-west orientation corresponds to a bearing of 315°.

Reply:
> That would mean that the sense of "positive" rotation also changes with the handedness of the CO system, i.e. one would need other rotation matrices depending on the handedness of the system. 

Isn't that the whole definition of what handedness of the CO system means?

So, spelled out, if you think about a rotation around the z-axis from the perspective of viewing from the origin towards the positive end of the z-axis, you have:

- Right-Handed Coordinate System: positive rotation is counterclockwise.
```math
$$
R_z^{\text{right}}(\alpha) = \begin{pmatrix}
\cos\alpha & -\sin\alpha & 0 \\
\sin\alpha & \cos\alpha & 0 \\
0 & 0 & 1
\end{pmatrix}
$$
```

- Left-Handed Coordinate System: Positive rotation is clockwise.

```math
R_z^{\text{left}}(\alpha) = \begin{pmatrix}
\cos\alpha & \sin\alpha & 0 \\
-\sin\alpha & \cos\alpha & 0 \\
0 & 0 & 1
\end{pmatrix}
```

For any arbitrary rotation around axis $\mathbf{u} = (u_x, u_y, u_z)$ by an angle $\theta$, the rotation matrix is:
- For Right-Handed Coordinate System:
```math
R_{\text{right}} = \begin{pmatrix}
\cos\theta + u_x^2 (1 - \cos\theta) & u_x u_y (1 - \cos\theta) - u_z \sin\theta & u_x u_z (1 - \cos\theta) + u_y \sin\theta \\
u_y u_x (1 - \cos\theta) + u_z \sin\theta & \cos\theta + u_y^2 (1 - \cos\theta) & u_y u_z (1 - \cos\theta) - u_x \sin\theta \\
u_z u_x (1 - \cos\theta) - u_y \sin\theta & u_z u_y (1 - \cos\theta) + u_x \sin\theta & \cos\theta + u_z^2 (1 - \cos\theta)
\end{pmatrix}
```

- For Left-Handed Coordinate System:
```math
R_{\text{left}} = \begin{pmatrix}
\cos\theta + u_x^2 (1 - \cos\theta) & u_x u_z (1 - \cos\theta) + u_y \sin\theta & u_x u_y (1 - \cos\theta) - u_z \sin\theta \\
u_z u_x (1 - \cos\theta) - u_y \sin\theta & \cos\theta + u_z^2 (1 - \cos\theta) & u_z u_y (1 - \cos\theta) + u_x \sin\theta \\
u_y u_x (1 - \cos\theta) + u_z \sin\theta & u_y u_z (1 - \cos\theta) - u_x \sin\theta & \cos\theta + u_y^2 (1 - \cos\theta)
\end{pmatrix}
```
This is also what @domna implemented in the nexus3d package [here](https://github.com/domna/nexus3d/blob/0a7899bf2556ef593488fa7468c6f6d9eff7708e/nexus3d/matrix.py#L10). Wouldn't that have to be the rotations that we represent by the vectors in NXtransformations? I think that was Florian's point yesterday that these should actually be part of the docstring if we want to allow both left- and right-handed CSs and rotations. There, we should probably also state the orientation that we choose (namely, we are looking from the origin towards positive axis values. This is also in line with what is the current definition of the [NeXus coordinate system](https://manual.nexusformat.org/design.html#the-nexus-coordinate-system))

See also [this](https://www.evl.uic.edu/ralph/508S98/coordinates.html) and [this](https://butterflyofdream.wordpress.com/2016/07/05/converting-rotation-matrices-of-left-handed-coordinate-system/) link.

Sorry for the long text, but I think it is worth it for us to think this through so we can make the correct statements also in the base classes.

Reply:
I guess for now, I could just set the sign of the angles to -1 because [NXtransformations/@vector](https://fairmat-nfdi.github.io/nexus_definitions/classes/base_classes/NXtransformations.html#nxtransformations-axisname-vector-attribute) explicitly mentions "right-handed rotation with increasing angle". In the future, we should then update that docstring to show what the vector means in a left-handed CS.

Reply:
Didn't know this, but according to the links you send this makes sense. But then I agree, even more reason to build in proper mechanism for defining the handedness, bases, and following rotation matrices.